src/Networking/NexusMods.Networking.NexusWebApi/NexusModsLibrary.Collections.cs

```diff
@@ -61,11 +61,73 @@ public partial class NexusModsLibrary
     }
 
     /// <summary>
-    /// Deletes a collection, all revisions, and all download entities of all revisions.
+    /// Gets all revision numbers that are newer than the current revision.
+    /// </summary>
+    public async ValueTask<RevisionNumber[]> GetNewerRevisionNumbers(
+        CollectionRevisionMetadata.ReadOnly currentRevision,
+        CancellationToken cancellationToken)
+    {
+        var revisionNumbers = await GetAllRevisionNumbers(currentRevision.Collection, cancellationToken);
+
+        var currentValue = currentRevision.RevisionNumber.Value;
+        var newer = revisionNumbers
+            .TakeWhile(other => other.Value > currentValue)
+            .ToArray();
+
+        return newer;
+    }
+
+    /// <summary>
+    /// Gets all revision numbers for the given collection in descending order.
     /// </summary>
-    public async ValueTask DeleteCollection(
-        CollectionMetadataId collectionMetadataId,
+    public async ValueTask<RevisionNumber[]> GetAllRevisionNumbers(
+        CollectionMetadata.ReadOnly collection,
         CancellationToken cancellationToken)
+    {
+        var gameDomain = await _mappingCache.TryGetDomainAsync(collection.GameId, cancellationToken);
+
+        var apiResult = await _gqlClient.CollectionRevisionNumbers.ExecuteAsync(
+            slug: collection.Slug.Value,
+            domainName: gameDomain.Value.Value,
+            viewAdultContent: true,
+            cancellationToken: cancellationToken
+        );
+
+        var revisions = apiResult.Data?.Collection.Revisions;
+        if (revisions is null) throw new NotSupportedException($"API call returned no data for collection slug `{collection.Slug}`");
+
+        var revisionNumbers = revisions
+            .Select(static x => x.RevisionNumber)
+            .OrderDescending()
+            .Select(static number => RevisionNumber.From((ulong)number))
+            .ToArray();
+
+        return revisionNumbers;
+    }
+
+    /// <summary>
+    /// Deletes a revision and all downloaded entities.
+    /// </summary>
+    public async ValueTask DeleteRevision(CollectionRevisionMetadataId revisionId)
```

Comment:
I'm a bit confused as to why `DeleteRevision` and `DeleteCollection` methods are here in `NexusMods.Networking.NexusWebApi`, wouldn't it make more sense for this stuff to live somewhere in the DataModel or something?

Reply:
`NexusModsLibrary` has somewhat become the place for all these operations.

Reply:
I don't want to necessarily fix that in this PR, but would you agree that it shouldn't be that way?

Reply:
They could go into `CollectionDownloader`

Reply:
Want to move in separate PR?